### ISSUE TITLE
[Networks app] Improvements to API

### ISSUE THREAD

**Issue by jamesscottbrown** [ove/ove-apps#78]
## Handling neighbors

In the networks app, `/operation/neighborsOf` currently accepts the name of single node, and applies to filter to hide nodes that are not connected to that node.

This has two main limitations:
- it is impossible to change the color of, or add labels for, the neighbors of a node: it is only possible to hide the nodes that are not neighbors
- it is impossible to select the neighbors of a set of nodes [this can't even be done in multiple stages, as selecting the neighbors of one node will hide ay previously selected nodes]

I think that `neighborsOf` should conceptually be an operation that simply takes a selection (which may consist of multiple nodes) and creates a new selection from it, without itself updating the display.
 This will require iterating over the nodes in the original selection, as the `neighborsOf` function provided by `sigma.plugins.filter` takes a single node name as an argument ([this is the function](https://github.com/jacomyal/sigma.js/blob/4ce817bd6f66f24f56a94965afdbebe5226ef9f6/plugins/sigma.plugins.filter/sigma.plugins.filter.js#L109) does the actual filtering).

However, rather than adding this as an operator to our query language, I suggest instead adding an additional parameter to the `search`/`color`/`labelNodes` operations to indicate whether they apply to the nodes matching the filter expression, their neighbours, or both. 

## Name of filtering operation

Currently the filtering operation is accessed through the API route `/operation/search`. However, what this operation really does is filter the display, so `filter` would be a more descriptive name than `search`.

*Every* operation first perform a 'search' before updating the display, so this is not a good name for any of them.

**Comment by senakafdo** [comment]
## Handling neighbors
I will look into improving the `neighborsOf` operation. 

What I have done there is to expose a function that is exposed by Sigma. In theory, you could do whatever you want with it, as displaying neighbors or neighbors or going 4 - 5 levels deep with it, but none of that is something they support and all of that would add to the complexity of this. But, your points on not being able to do anything about the neighbors beyond simply displaying them and hiding the rest is valid.

## Name of filtering operation
The choice of `search` instead of `filter` was deliberate.

You can use a `filter` to colour or label the nodes as well. Similarly, you can use a `filter` to search for something. This follows the [OData Specification](https://www.odata.org/getting-started/basic-tutorial/#filter) If you call the operation `filter` as well, everything starts to become a bit confusing. 

I don't see a big issue in terms of naming the operation `search` or `filter`, because the user will interpret either in the correct sense in this context. However, the choice of the operation name `filter` with few other operations accepting parameters called `filter` does not sound right.

> However, what this operation really does is filter the display, so filter would be a more descriptive name than search.

Not exactly, it searches for something using a filter which can be for nodes or edges depending on the parameters you have passed. And, before this you would have loaded the graph in some format or queried the database. From a Sigma.js point of view as well, yet again, I'm using their filter plugin for all operations, be it search, colour or displaying a label.

In other words, whatever operation you perform, search, colour or display a label, you are filtering the dataset. So, it does not make sense to have `filter` as the name of the `search` operation.

So, everything actually does make sense, in my opinion. Therefore I will stick to my choice of naming, in this case.

**Comment by senakafdo** [comment]
As chatted offline we can rename `search` to `showOnly`. Still need to agree on what to do with `neighborsOf`.

**Comment by jamesscottbrown** [comment]
Another current limitation of this API is that all filtering/highlighting operations act on all network sections.

If, for example, different people are viewing networks on the GDO and DODev, neither can filter their view without affecting the other.

**Comment by senakafdo** [comment]
The request to change `search` is done.

I have reported jacomyal/sigma.js#1009 and parked this issue. Should there be a pressing need for a better way of filtering neighbors later on, or if jacomyal/sigma.js#1009 is resolved, we can reopen and fix this issue.

**Comment by senakafdo** [comment]
Resolving issue as there are no signs of jacomyal/sigma.js#1009 being fixed.
